modules/react/select/lib/hooks/useSelectInput.ts

```diff
@@ -0,0 +1,209 @@
+import React from 'react';
+import {composeHooks, createElemPropsHook} from '@workday/canvas-kit-react/common';
+import {useComboboxInput} from '@workday/canvas-kit-react/combobox';
+import {useSelectModel} from './useSelectModel';
+
+/**
+ * `useSelectInput` extends `useComboboxInput` while adding type ahead functionality and adding select specific [keyboard support](https://www.w3.org/WAI/ARIA/apg/patterns/combobox/examples/combobox-select-only/)
+ */
+export const useSelectInput = composeHooks(
+  useComboboxInput,
+  createElemPropsHook(useSelectModel)((model, ref) => {
+    const keySofar = React.useRef('');
+    const timer = React.useRef<ReturnType<typeof setTimeout>>();
+
+    React.useLayoutEffect(() => {
+      model.events.setWidth(model.state.targetRef.current?.clientWidth || 0);
```

Comment:
Before the popup is open, set the width to the the width of the input. In a case where the consumer sets `initialVisibility` to `visible` for some odd reason, the popup will have the width and the items will be visible.